qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -0,0 +1,605 @@
+                "rise", "rms", "rmsjitter", "rmsnoise", "sigma1",
+                "sigma2", "sigma3", "sixsigmajit", "snratio", "stddev",
+                "undefined", "waveforms"
+            ),
+            docstring="Please see page 566-569 of the programmers manual "
+                      "for a detailed description of these arguments. "
+                      "http://download.tek.com/manual/077001022.pdf"
+        )
+
+        self.add_parameter(
+            "unit",
+            get_cmd=f"MEASUrement:MEAS{self._measurement_number}:UNIts?",
+            get_parser=strip_quotes
+        )
+
+        for src in [1, 2]:
+            self.add_parameter(
+                f"source{src}",
+                get_cmd=f"MEASUrement:MEAS{self._measurement_number}:SOUrce{src}?",
+                set_cmd=partial(self._set_source, src),
+                vals=Enum(
+                    *[f"CH{i}" for i in range(1, TektronixDPO7000xx.channel_count + 1)]
+                ),
+            )
+
+    def _set_measurement_type(self, value):
+        self._adjustment_time = time.time()
+        self.write(
+            f"MEASUrement:MEAS{self._measurement_number}:TYPe {value}"
+        )
+
+    def _set_source(self, source_number, value):
+        self._adjustment_time = time.time()
+        self.write(
+            f"MEASUrement:MEAS{self._measurement_number}:SOUrce{source_number} {value}"
+        )
+
+    @property
+    def value(self) -> Parameter:
+        """
+        Return the appropriate parameter for the selected measurement
+        type
+        """
+        measurement_type = self.type()
+        name = f"_{measurement_type}_measurement"
+
+        if name not in self.parameters:
+            self.add_parameter(
+                name,
+                get_cmd=self._measure,
+                get_parser=float,
+                unit=self.unit()
+            )
+
+        return self.parameters[name]
+
+    def _measure(self) -> Any:
+        """
+        We need to wait a minimum amount of time after performing
+        some set actions to get a measurement value. Note that we
+        cannot use the post_delay or inter_delay parameter options
+        here, because these are minimum delays between consecutive
+        set operations, not delays between set and get of two
+        different parameters.
+        """
+
+        time_since_adjust = time.time() - self._adjustment_time
+        if time_since_adjust < self.minimum_adjustment_time:
+            time_remaining = self.minimum_adjustment_time - time_since_adjust
+            time.sleep(time_remaining)
+
+        return self.ask(f"MEASUrement:MEAS{self._measurement_number}:VALue?")
+
+
+class TektronixDPO7000xx(VisaInstrument):
+    """
+    QCoDeS driver for the MSO/DPO5000/B, DPO7000/C,
+    DPO70000/B/C/D/DX/SX, DSA70000/B/C/D, and
+    MSO70000/C/DX Series Digital Oscilloscopes
+    """
+    channel_count = 4
+    measurement_count = 8
```

Comment:
may i suggest to rename these to `number_of_*`?

Reply:
Ok